pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -0,0 +1,196 @@
+import io
+import os
+from typing import Any, Dict, Iterable, Optional, Sequence, Union
+
+import torch
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins.training_type.ddp_spawn import DDPSpawnPlugin
+from pytorch_lightning.plugins.training_type.utils import on_colab_kaggle
+from pytorch_lightning.utilities import _TPU_AVAILABLE, rank_zero_warn
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.seed import seed_everything
+
+if _TPU_AVAILABLE:
+    import torch_xla.core.xla_model as xm
+    import torch_xla.distributed.parallel_loader as xla_pl
+    import torch_xla.distributed.xla_multiprocessing as xmp
+    from torch_xla.core.xla_model import rendezvous
+    from torch_xla.distributed.parallel_loader import ParallelLoader
+else:
+    xm, xla_pl, xmp, ParallelLoader, rendezvous = [None] * 5
+
+
+class TPUSpawnPlugin(DDPSpawnPlugin):
+
+    def __init__(self, parallel_devices: Sequence[int], num_nodes: int = 1, **kwargs: Dict[str, Any]) -> None:
+
+        super().__init__(
+            parallel_devices, num_nodes=num_nodes, cluster_environment=None, sync_batchnorm=False, **kwargs
+        )
+        self.tpu_local_core_rank = 0
+        self.start_method = None
+
+    @property
+    def distributed_sampler_kwargs(self) -> dict:
+        return dict(num_replicas=xm.xrt_world_size(), rank=xm.get_ordinal())
+
+    def process_dataloader(self, dataloader: Union[Iterable, torch.utils.data.DataLoader]) -> ParallelLoader:
+        device = xm.xla_device()
+        dataloader = xla_pl.ParallelLoader(dataloader, [device])
+        dataloader = dataloader.per_device_loader(device)
+        return dataloader
+
+    def configure_ddp(self) -> None:
+        pass
+
+    def init_ddp_connection(self, global_rank: int, world_size: int) -> None:
+        pass
+
+    def set_world_ranks(self, process_idx: int) -> None:
+        self.tpu_local_core_rank = xm.get_local_ordinal()
+        self.tpu_global_core_rank = xm.get_ordinal()
+        self.global_rank = self.tpu_local_core_rank
+        self.world_size = self.num_nodes * self.num_processes
+
+    def new_process(self, process_idx: int, trainer) -> None:
```

Comment:
```suggestion
    def create_new_process(self, process_idx: int, trainer) -> None:
```

Reply:
We should not do this, since this is the name all parallel plugins use. Also it does not create the new process, but this is the function running in the new project.